x-pack/plugins/security_solution/public/cases/components/create/form_context.test.tsx

```diff
@@ -0,0 +1,470 @@
+  const onFormSubmitSuccess = jest.fn();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    usePostCaseMock.mockImplementation(() => defaultPostCase);
+    useConnectorsMock.mockReturnValue(sampleConnectorData);
+    useCaseConfigureMock.mockImplementation(() => useCaseConfigureResponse);
+    useGetIncidentTypesMock.mockReturnValue(useGetIncidentTypesResponse);
+    useGetSeverityMock.mockReturnValue(useGetSeverityResponse);
+    useGetIssueTypesMock.mockReturnValue(useGetIssueTypesResponse);
+    useGetFieldsByIssueTypeMock.mockReturnValue(useGetFieldsByIssueTypeResponse);
+
+    (useGetTags as jest.Mock).mockImplementation(() => ({
+      tags: sampleTags,
+      fetchTags,
+    }));
+  });
+
+  describe('Step 1 - Case Fields', () => {
+    it('it renders', async () => {
+      const wrapper = mount(
+        <TestProviders>
+          <FormContext onSuccess={onFormSubmitSuccess}>
+            <CreateCaseForm />
+            <SubmitCaseButton />
+          </FormContext>
+        </TestProviders>
+      );
+
+      expect(wrapper.find(`[data-test-subj="caseTitle"]`).first().exists()).toBeTruthy();
+      expect(wrapper.find(`[data-test-subj="caseDescription"]`).first().exists()).toBeTruthy();
+      expect(wrapper.find(`[data-test-subj="caseTags"]`).first().exists()).toBeTruthy();
+      expect(wrapper.find(`[data-test-subj="caseConnectors"]`).first().exists()).toBeTruthy();
+      expect(
+        wrapper.find(`[data-test-subj="case-creation-form-steps"]`).first().exists()
+      ).toBeTruthy();
+    });
+
+    it('should post case on submit click', async () => {
+      useConnectorsMock.mockReturnValue({
+        ...sampleConnectorData,
+        connectors: connectorsMock,
+      });
+
+      const wrapper = mount(
+        <TestProviders>
+          <FormContext onSuccess={onFormSubmitSuccess}>
+            <CreateCaseForm />
+            <SubmitCaseButton />
+          </FormContext>
+        </TestProviders>
+      );
+
+      await fillForm(wrapper);
+      wrapper.update();
+
+      await act(async () => {
+        wrapper.find(`[data-test-subj="create-case-submit"]`).first().simulate('click');
+      });
+      await waitFor(() => expect(postCase).toBeCalledWith(sampleData));
+    });
+
+    it('should toggle sync settings', async () => {
+      useConnectorsMock.mockReturnValue({
+        ...sampleConnectorData,
+        connectors: connectorsMock,
+      });
+
+      const wrapper = mount(
+        <TestProviders>
+          <FormContext onSuccess={onFormSubmitSuccess}>
+            <CreateCaseForm />
+            <SubmitCaseButton />
+          </FormContext>
+        </TestProviders>
+      );
+
+      await fillForm(wrapper);
+
+      act(() => {
+        wrapper.find('[data-test-subj="caseSyncAlerts"] button').first().simulate('click');
+        wrapper.update();
+      });
+
+      await act(async () => {
+        wrapper.find(`[data-test-subj="create-case-submit"]`).first().simulate('click');
+      });
```

Comment:
both of the `act`s and the `update` can be dropped
```suggestion
   wrapper.find('[data-test-subj="caseSyncAlerts"] button').first().simulate('click');
   wrapper.find(`[data-test-subj="create-case-submit"]`).first().simulate('click');
```

Reply:
For one test the act is needed to be passed. We need to `await` for all promises to be fulfilled.